src/vconsole/vconsole-setup.c

```diff
@@ -328,48 +320,88 @@ static void setup_remaining_vcs(int fd, bool utf8) {
         }
 }
 
+static int find_source_vc(char **ret_path, int *ret_idx, const char *src_vc) {
+        int ret_fd;
+
+        if (src_vc) {
+                _cleanup_close_ int fd = -1;
+
+                if (!(*ret_path = malloc(strlen(src_vc) + 1))) {
```

Comment:
hmm, please do not clobber return parameters on failure (see CODING_STYLE). i.e. a function should only overwrite return parameters on success, but on failure leave them the way they are.

also, we generally prefer using type-safe `new(char, strlen(src_vc) + 1)` instead of malloc() when allocating place for strings.

Finally, we really try to make things readable, and hence much prefer if each line does one thing. Specifically, I think this should be two lines: one line doing the allocation, and the next line checking if it worked. (also see CODING_STYLE)